### ISSUE TITLE
minor refactoring of WarehouseSpec

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#954]
[This thread](https://github.com/akuity/kargo/pull/911#discussion_r1357491853) suggests some minor refactoring of the `WarehouseSpec` and we agreed to tackle it as a follow-up.

Making a separate issue so we don't lose track of this after #911.
